src/herbpy/action/blocks.py

```diff
@@ -0,0 +1,172 @@
+import logging, numpy, openravepy, prpy
+from prpy.action import ActionMethod
+from prpy.planning.base import PlanningError
+from prpy.util import ComputeEnabledAABB
+
+logger = logging.getLogger('herbpy')
+
+class NoTSRException(Exception):
+    pass
+
+def _GrabBlock(robot, blocks, table, manip=None, preshape=[1.7, 1.7, 0.2, 2.45],
+              **kw_args):
+    """
+    @param robot The robot performing the grasp
+    @param block The block to grab
+    @param table The table, or object, the block is resting on
+    @param manip The manipulator to use to grab the block
+      (if None active manipulator is used)
+    """
+    from prpy.rave import AllDisabled, Disabled
+    from prpy.viz import RenderTSRList, RenderVector
+
+    env = robot.GetEnv()
+    block = None
+
+    if manip is None:
+        manip = robot.GetActiveManipulator()
+    
+    # First move the hand to the right preshape
+    manip.hand.MoveHand(*preshape)
+
+    block_tsr_list = []
+    for b in blocks:
+        # Get a TSR to move near the block.
+        tsr_list = robot.tsrlibrary(b, 'grasp', manip=manip)
+        block_tsr_list += tsr_list
```

Comment:
Hmm...I thought we actually decided the opposite. We have been adding locks to the tsrlibrary wherever we find missing ones. I am not opposed to going with this paradym instead, but I think we should do it in a separate issue/PR where we carefully go through the library and remove or cleanup locking.